.spin/cmds.py

```diff
@@ -0,0 +1,500 @@
+    ctx.params[MESON_ARGS]["install"] = tuple()
+    ctx.params[MESON_ARGS]["setup"] = meson_args_
+
+    if sys.platform == "cygwin":
+        # Cygwin only has netlib lapack, but can link against
+        # OpenBLAS rather than netlib blas at runtime.  There is
+        # no libopenblas-devel to enable linking against
+        # openblas-specific functions or OpenBLAS Lapack
+        ctx.params[MESON_ARGS]["setup"] = ctx.params[MESON_ARGS]["setup"] + ("-Dlapack=lapack", "-Dblas=blas")
+
+    if ctx.params['werror']:
+        ctx.params[MESON_ARGS]["setup"] = ctx.params[MESON_ARGS]["setup"] + ("--werror", )
+
+    if ctx.params['debug'] or ctx.params['release']:
+        if ctx.params['debug'] and ctx.params['release']:
+            raise ValueError("Set at most one of `--debug` and `--release`!")
+        if ctx.params['debug']:
+            buildtype = 'debug'
+            cflags_unwanted = ('-O1', '-O2', '-O3')
+        elif ctx.params['release']:
+            buildtype = 'release'
+            cflags_unwanted = ('-O0', '-O1', '-O2')
+        ctx.params[MESON_ARGS]["setup"] = ctx.params[MESON_ARGS]["setup"] + (f"-Dbuildtype={buildtype}", )
+        if 'CFLAGS' in os.environ.keys():
+            # Check that CFLAGS doesn't contain something that supercedes -O0
+            # for a plain debug build (conda envs tend to set -O2)
+            cflags = os.environ['CFLAGS'].split()
+            for flag in cflags_unwanted:
+                if flag in cflags:
+                    raise ValueError(f"A {buildtype} build isn't possible, "
+                                        f"because CFLAGS contains `{flag}`."
+                                        "Please also check CXXFLAGS and FFLAGS.")
+    if ctx.params['gcov']:
+        ctx.params[MESON_ARGS]["setup"] = ctx.params[MESON_ARGS]["setup"] + ('-Db_coverage=true', )
+
+    if ctx.params['asan']:
+        ctx.params[MESON_ARGS]["setup"] = ctx.params[MESON_ARGS]["setup"] + ('-Db_sanitize=address,undefined', )
+
+    if ctx.params['setup_args']:
+        ctx.params[MESON_ARGS]["setup"] = ctx.params[MESON_ARGS]["setup"] + tuple([str(arg) for arg in args.setup_args])
+
+    if ctx.params['with_accelerate']:
+        # on a mac you probably want to use accelerate over scipy_openblas
+        ctx.params[MESON_ARGS]["setup"] = ctx.params[MESON_ARGS]["setup"] + ("-Dblas=accelerate", )
+    elif ctx.params['with_scipy_openblas']:
+        configure_scipy_openblas()
+        os.env['PKG_CONFIG_PATH'] = os.pathsep.join([
+                os.getcwd(),
+                os.env.get('PKG_CONFIG_PATH', '')
+                ])
+
+    if ctx.params['parallel'] is None:
+        # Use number of physical cores rather than ninja's default of 2N+2,
+        # to avoid out of memory issues (see gh-17941 and gh-18443)
+        n_cores = cpu_count(only_physical_cores=True)
+        ctx.params[MESON_ARGS]["jobs"] = n_cores
+    else:
+        ctx.params[MESON_ARGS]["jobs"] = ctx.parallel
```

Comment:
Here, when testing on MacOS ARM M3, `spin build -j 16 --with-accelerate` fails with:

<details>

```
Traceback (most recent call last):
  File "/Users/treddy/python_venvs/py_312_scipy_dev/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/treddy/python_venvs/py_312_scipy_dev/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/treddy/github_projects/scipy/.spin/cmds.py", line 385, in build
    ctx.params[MESON_ARGS]["jobs"] = ctx.parallel
                                     ^^^^^^^^^^^^
AttributeError: 'Context' object has no attribute 'parallel'

An internal error has occurred. Please file a bug report at

  https://github.com/scientific-python/spin

including the above traceback and the following information:

  spin: 0.12, package: scipy

Aborting.
```

</details>

Simpler incantations also seem to choke on MacOS ARM:

`spin test -j 16`

```
$ meson setup build --prefix=/usr compile install setup jobs
usage: meson [-h] {setup,configure,dist,install,introspect,init,test,wrap,subprojects,rewrite,compile,devenv,env2mfile,format,fmt,help} ...
meson: error: unrecognized arguments: compile install setup jobs
```

Reply:
Regarding this - `spin build -j 16 --with-accelerate` works perfectly fine on my Apple M2 MacBook Pro. `spin test -j 16` also works fine. However note that you need to follow the steps I have listed in https://github.com/scipy/scipy/pull/21674#issue-2571022156 (see under **Additional Information**).

Reply:
N.B. - https://github.com/scipy/scipy/pull/21674#issuecomment-2465092245